yarn.lock

```diff
@@ -32089,13 +32303,20 @@ __metadata:
   languageName: node
   linkType: hard
 
-"regenerator-runtime@npm:^0.11.0, regenerator-runtime@npm:^0.13.4, regenerator-runtime@npm:^0.13.7":
```

Comment:
This is reference to the above comment about version resolutions — hopefully this explains things better. `regenerator-runtime` is an example of a package that we are manually resolving with `"**/regenerator-runtime": "^0.13.7"`. If I understand correctly  this means "any time `regenerator-runtime` appears, ensure that `^0.13.7` is installed" (which resolved to 0.13.7 when `yarn` was run last, hence why it says `version: 0.13.7` in this section). However, I'm concerned that with this commit, that resolution, and other similar manual resolutions, are getting overridden. It's difficult for me to suggest a solution, as I don't really understand how Yarn works in this regard, but I'm wondering if 1) this is really a concern and 2) if it is, is there a way that we can test this?

Reply:
```
➜ yarn info -A -R regenerator-runtime --dependents
├─ regenerator-runtime@npm:0.13.7
│  └─ Version: 0.13.7
│
└─ regenerator-runtime@patch:regenerator-runtime@npm%3A0.13.7#./.yarn/patches/regenerator-runtime-npm-0.13.7-41bcbe64ea.patch::version=0.13.7&hash=16e0d6&locator=metamask-crx%40workspace%3A.
   ├─ Version: 0.13.7
   │
   └─ Dependents
      ├─ @babel/runtime-corejs3@npm:7.11.2
      ├─ @babel/runtime@npm:7.18.9
      ├─ @babel/runtime@patch:@babel/runtime@npm%3A7.18.9#./.yarn/patches/@babel-runtime-npm-7.18.9-28ca6b5f61.patch::version=7.18.9&hash=918bda&locator=metamask-crx%40workspace%3A.
      ├─ @storybook/addon-a11y@npm:6.5.10
      ├─ @storybook/addon-a11y@npm:6.5.10 [15918]
      ├─ @storybook/addon-actions@npm:6.5.10
      ├─ @storybook/addon-actions@npm:6.5.10 [15918]
      ├─ @storybook/addon-backgrounds@npm:6.5.10
      ├─ @storybook/addon-backgrounds@npm:6.5.10 [8ce55]
      ├─ @storybook/addon-docs@npm:6.5.10
      ├─ @storybook/addon-docs@npm:6.5.10 [8ce55]
      ├─ @storybook/addon-essentials@npm:6.5.10
      ├─ @storybook/addon-essentials@npm:6.5.10 [15918]
      ├─ @storybook/addon-outline@npm:6.5.10
      ├─ @storybook/addon-outline@npm:6.5.10 [8ce55]
      ├─ @storybook/addon-toolbars@npm:6.5.10
      ├─ @storybook/addon-toolbars@npm:6.5.10 [8ce55]
      ├─ @storybook/addon-viewport@npm:6.5.10
      ├─ @storybook/addon-viewport@npm:6.5.10 [8ce55]
      ├─ @storybook/addons@npm:6.5.10
      ├─ @storybook/addons@npm:6.5.10 [15918]
      ├─ @storybook/addons@npm:6.5.10 [90fde]
      ├─ @storybook/api@npm:6.5.10
      ├─ @storybook/api@npm:6.5.10 [15918]
      ├─ @storybook/api@npm:6.5.10 [6515a]
      ├─ @storybook/client-api@npm:6.5.10
      ├─ @storybook/client-api@npm:6.5.10 [15918]
      ├─ @storybook/components@npm:6.5.10
      ├─ @storybook/components@npm:6.5.10 [15918]
      ├─ @storybook/core-client@npm:6.5.10
      ├─ @storybook/core-client@npm:6.5.10 [5dda7]
      ├─ @storybook/core-client@npm:6.5.10 [9f512]
      ├─ @storybook/core-client@npm:6.5.10 [ae428]
      ├─ @storybook/core-server@npm:6.5.10
      ├─ @storybook/core-server@npm:6.5.10 [9f512]
      ├─ @storybook/csf-tools@npm:6.5.10
      ├─ @storybook/csf-tools@npm:6.5.10 [5dda7]
      ├─ @storybook/docs-tools@npm:6.5.10
      ├─ @storybook/manager-webpack4@npm:6.5.10
      ├─ @storybook/manager-webpack4@npm:6.5.10 [5dda7]
      ├─ @storybook/preview-web@npm:6.5.10
      ├─ @storybook/preview-web@npm:6.5.10 [19b21]
      ├─ @storybook/react@npm:6.5.10
      ├─ @storybook/react@npm:6.5.10 [15918]
      ├─ @storybook/router@npm:6.5.10
      ├─ @storybook/router@npm:6.5.10 [6515a]
      ├─ @storybook/router@npm:6.5.10 [72474]
      ├─ @storybook/source-loader@npm:6.5.10
      ├─ @storybook/source-loader@npm:6.5.10 [19b21]
      ├─ @storybook/store@npm:6.5.10
      ├─ @storybook/store@npm:6.5.10 [06089]
      ├─ @storybook/store@npm:6.5.10 [aa3b3]
      ├─ @storybook/telemetry@npm:6.5.10
      ├─ @storybook/theming@npm:6.5.10
      ├─ @storybook/theming@npm:6.5.10 [15918]
      ├─ @storybook/theming@npm:6.5.10 [6515a]
      ├─ @storybook/ui@npm:6.5.10
      ├─ @storybook/ui@npm:6.5.10 [f9051]
      └─ babel-runtime@npm:6.26.0
```

all dependents are targeting 0.13.7 and the patched file.

Reply:
Can you provide me more info as to why you are concerned they are being overridden?

Reply:
#16242  < - restores the symbol observable resolution

Reply:
Okay, thanks. My concern purely comes from what I'm seeing in `yarn.lock`. It appears, based on the diff, that previously Yarn was configured to resolve multiple versions of `@babel/runtime` to the same version, and now that's no longer happening. In other words, instead of `^7.0.0`, `^7.10.2`, etc. resolving to 7.18.9, now only `7.18.9` resolves to 7.18.9, and then there's a patch that targets 7.18.9. This introduces a risk: if, for instance, one of our dependencies declares `@babel/runtime` as its dependency and it uses `^7.10.2` as the version specifier, 7.10.2 could get installed and not 7.18.9.

That said, looking at it again, I think the risk is very low and my concern is unfounded. First, all of the previous version specifiers (`^7.0.0`, `^7.10.2`, etc.) already resolve to 7.18.9, so I don't think the resolution logic that we specified before is changing. The only version of `@babel/runtime` that seems to have been installed before was 7.18.9, and I think that would remain the same. Second, I checked out your `yarn3-activate` branch and it looks like the resolution for `@babel/runtime` has been removed from the `resolutions` field in `package.json` altogether. In fact there is no package referred to by any of the files in `patches` that is also in `resolutions`.

All this to say... it looks to be safe to add all of the existing patches as resolutions as you are doing. This _should not_ result in any different versions of packages getting installed.

Reply:
Cool. once all the different branches merge into the main one we can re-evaluate the differences in resolved versions and make amends before it lands in develop